mlir/lib/Dialect/XeGPU/Transforms/XeGPUUnroll.cpp

```diff
@@ -575,7 +577,8 @@ struct UnrollStoreScatterOp : public UnrollPattern<xegpu::StoreScatterOp> {
     VectorType valueTy = llvm::dyn_cast<VectorType>(op.getValue().getType());
     xegpu::TensorDescType tdescTy = op.getTensorDescType();
 
-    if (!tdescTy.isScattered())
+    // TODO: handle the unstructure source case (!tdesTy)
+    if (!tdescTy || !tdescTy.isScattered())
```

Comment:
nit: I think the `isScattered` is already checked by op verifier and can be taken as an invariant in the passes